packages/components/src/components/PlaygroundSettings.tsx

```diff
@@ -149,6 +137,11 @@ export function generateFunctionPlotSettings(settings: PlaygroundSettings) {
   return { xScale, points: settings.functionChartSettings.count };
 }
 
+// settings for the settings form
+export type MetaSettings = {
+  disableLogX?: boolean;
```

Comment:
Btw, I think that doing this is overkill at this point. The overrides are likely not super used.

Reply:
There's a long story behind this.

In this PR, I only extracted the option from `PlaygroundSettings`; it existed previously through `fixed`, but `fixed` doesn't exist in the new form components, and passing it around lead to more complicated code than necessary.

Previously it was done with a secret settings field, which itself was an artifact of scales refactoring (before scales, it was a checkbox that was disabled through `fixed` setting).

I'd probably go with a single `disableLogX` prop instead of `MetaSettings` here, and I almost did so, but I remember that a few months I changed this in the opposite direction, so I won't want another back-and-forth on this.